### PR TITLE
feat(expr): Implement approx_count_distinct for batch queries

### PR DESCRIPTION
**Pull request by Graphcalibur**
## What's changed and what's your intention?

* Implement `approx_count_distinct` for batch queries by implementing the HyperLogLog algorithm outlined [in this paper](https://storage.googleapis.com/pub-tools-public-publication-data/pdf/40671.pdf)
* `approx_count_distinct` in stream processing will be handled in a separate PR.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#2727

### CODE REVIEW

File: src/expr/src/vector_op/agg/approx_count_distinct.rs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        answer as i64
+    }
+}
+
+impl Aggregator for ApproxCountDistinct {
+    fn return_type(&self) -> DataType {
+        self.return_type.clone()
+    }
+
+    fn update_with_row(&mut self, input: &DataChunk, row_id: usize) -> Result<()> {
```
Comment (fuyufjh) [inline src/expr/src/vector_op/agg/approx_count_distinct.rs:128]: (Not related to this PR...) Why these functions of `Aggregator` returns `Result`? 😇

File: src/expr/src/vector_op/agg/approx_count_distinct.rs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    fn count_hash(&self, mut hash: u64) -> u8 {
+        let mut count = 1;
+
+        hash >>= INDEX_BITS; // Ignore bits used as index for the hash
+        hash |= 1 << COUNT_BITS; // To allow hash to terminate if it is all 0s
+
+        while hash & 1 == 0 {
+            count += 1;
+            hash >>= 1;
+        }
```
Comment (fuyufjh) [inline src/expr/src/vector_op/agg/approx_count_distinct.rs:80]: X86 supports a bit-scan instruction to do this efficiently, which can be called by `__builtin_ctz` in C++. Is it possible to use that in Rust?

**Review by fuyufjh — COMMENTED** [review]
Rest LGTM

**Review by lmatz — COMMENTED** [review]
LGTM!

I am wondering whether it is necessary to set up a simple bench/test in the future to collect statistics that can be used to draw some of the figures in the paper, so that if we add some variants later, they can be compared with each other.

**Review by fuyufjh — APPROVED** [review]

**Review by liurenjie1024 — APPROVED** [review]

**Pull request merged**